In [1]:
!pip install --upgrade pandas
!pip install --upgrade pandas-datareader

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/
     |████████████████████████████████| 109 kB 6.1 MB/s 
  Attempting uninstall: pandas-datareader
    Found existing installation: pandas-datareader 0.9.0
    Uninstalling pandas-datareader-0.9.0:
      Successfully uninstalled pandas-datareader-0.9.0


In [12]:
import numpy as np
import math
import pandas as pd
import pandas_datareader as pdr
import matplotlib
import matplotlib.pyplot as plt

AMD = pdr.get_data_yahoo('AMD',
start="01/01/2017", interval="m")
INTC = pdr.get_data_yahoo('INTC',
start="01/01/2017", interval="m")

PRIMERO OBTENEMOS LOS DATOS DE LAS EMPRESAS 'MICROSOFT' Y 'APPLE' PARA PODER COMPARAR SUS GANANCIAS

In [13]:
AMD

High        Low        Open       Close      Volume  \
Date                                                                    
2017-01-01   11.690000   9.420000   11.420000   10.370000   878890400   
2017-02-01   15.550000  10.810000   10.900000   14.460000  1571999700   
2017-03-01   15.090000  12.380000   15.080000   14.550000  1733593200   
2017-04-01   14.740000  12.220000   14.600000   13.300000   943383500   
2017-05-01   13.630000   9.850000   13.430000   11.190000  1716437000   
...                ...        ...         ...         ...         ...   
2022-05-01  104.550003  83.269997   85.660004  101.860001  2898334400   
2022-06-01  109.570000  75.480003  102.129997   76.470001  2030987200   
2022-07-01   94.809998  71.599998   75.190002   94.470001  1633201100   
2022-08-01  104.589996  93.620003   95.589996  100.199997  1100815900   
2022-08-17   99.639999  96.900002   99.260002   97.099899    23502293   

             Adj Close  
Date                    
2017-01-01   10.370000  
2017-02-01   14.460000  
2017-03-01   14.550000  
2017-04-01   13.300000  
2017-05-01   11.190000  
...                ...  
2022-05-01  101.860001  
2022-06-01   76.470001  
2022-07-01   94.470001  
2022-08-01  100.199997  
2022-08-17   97.099899  

[69 rows x 6 columns]

In [14]:
INTC

High        Low       Open      Close       Volume  Adj Close
Date                                                                          
2017-01-01  38.450001  36.189999  36.610001  36.820000  456335100.0  31.593966
2017-02-01  36.950001  34.840000  36.820000  36.200001  477551300.0  31.061970
2017-03-01  36.299999  34.660000  35.849998  36.070000  546151800.0  31.171377
2017-04-01  37.500000  35.209999  36.189999  36.150002  444122200.0  31.240513
2017-05-01  37.169998  35.020000  36.110001  36.110001  432927300.0  31.205940
...               ...        ...        ...        ...          ...        ...
2022-05-01  46.639999  40.310001  43.720001  44.419998  773236800.0  43.631149
2022-06-01  44.930000  36.599998  44.770000  37.410000  783353000.0  37.036106
2022-07-01  40.730000  35.240002  37.090000  36.310001  792839800.0  35.947102
2022-08-01  37.689999  34.400002  36.049999  36.189999  515751300.0  35.828297
2022-08-17  35.820000  35.310001  35.810001  35.345001    8562007.0  35.345001

[69 rows x 6 columns]

OBTENEMOS EL CAMBIO MENSUAL DE CADA COMPAÑÍA USANDO EL ADJ. CLOSE

In [15]:
AMD["r"] = np.log(AMD['Adj Close']) - np.log(AMD['Adj Close'].shift(1))
print(AMD["r"])

Date
2017-01-01         NaN
2017-02-01    0.332469
2017-03-01    0.006205
2017-04-01   -0.089827
2017-05-01   -0.172744
                ...   
2022-05-01    0.174849
2022-06-01   -0.286701
2022-07-01    0.211384
2022-08-01    0.058886
2022-08-17   -0.031428
Name: r, Length: 69, dtype: float64


In [16]:
INTC["r"] = np.log(INTC['Adj Close']) - np.log(INTC['Adj Close'].shift(1))
print(INTC["r"])

Date
2017-01-01         NaN
2017-02-01   -0.016982
2017-03-01    0.003516
2017-04-01    0.002215
2017-05-01   -0.001107
                ...   
2022-05-01    0.018862
2022-06-01   -0.163878
2022-07-01   -0.029845
2022-08-01   -0.003310
2022-08-17   -0.013581
Name: r, Length: 69, dtype: float64


#DECLARAMOS NUESTRAS HIPOTESIS


*   H0: INTC_MEAN - AMD_MEAN = 0
*   HA: INTC_MEAN - AMD_MEAN <> 0

In [17]:
AMD_r_mean = AMD["r"].mean()
AMD_r_mean

0.032894460690197413

In [18]:
INTC_r_mean = INTC["r"].mean()
INTC_r_mean

0.0016498652357474725

In [19]:
AMD_r_var = AMD["r"].var()
AMD_r_var

0.025925895337105594

In [20]:
INTC_r_var = INTC["r"].var()
INTC_r_var

0.005553196846820742

OBTENEMOS MEDIAS Y VARIANZAS DE AMBAS EMPRESAS PARA PODER USAR NUESTRA FORMULA DE T

In [22]:
t = (AMD_r_mean - INTC_r_mean - 0) / math.sqrt((INTC_r_var + AMD_r_var) / INTC["r"].count())
t

1.452172681859927

PODEMOS OBSERVAR QUE LA T ES 1.45, POR LO QUE NUESTRA HIPOTESIS NULA NO PUEDE SER DESCARTADA POR COMPLETO, PERO TAMBIEN VEMOS QUE NO ES 0, POR LO QUE SI HAY UNA DIFERENCIA, Y COMO NOSOTROS SUPUSIMOS QUE AMD_MEAN - INTC_MEAN <> 0  Y OBTUVIMOS UN VALOR POSITIVO, VEMOS QUE ESTO ES CIERTO Y EN ESTE CASO AL USAR:

 AMD_MEAN - INTC_MEAN > 0 => AMD_MEAN > INTC_MEAN